crates/swc_config/src/regex_js.rs

```diff
@@ -0,0 +1,42 @@
+use std::{ops::Deref, sync::Arc};
+
+pub use anyhow::Error;
+use anyhow::{Context, Result};
+use dashmap::DashMap;
+use once_cell::sync::Lazy;
+use regress::Regex;
+use rustc_hash::FxBuildHasher;
+
+#[derive(Debug, Clone)]
+pub struct CachedJsRegex {
+    regex: Arc<Regex>,
```

Comment:
Also, it may worth to store original string here, and export it using `fn source(&self) -> &str`.